src/Component/AutoMapper/Generator/Generator.php

```diff
@@ -139,13 +139,13 @@ public function generate(MapperGeneratorMetadataInterface $mapperGeneratorMetada
         foreach ($propertiesMapping as $propertyMapping) {
             $transformer = $propertyMapping->getTransformer();
 
-            if (\in_array($propertyMapping->getProperty(), $inConstructor, true)) {
-                continue;
-            }
```

Comment:
If we don't add setters here and we have a constructor with properties and a target to populate, we will skip the initialize part and none of the properties will be set, so the best is to duplicate the values to set by ignoring this condition.

I'm not 100% sure of that but it works on the whole test suite and fix my issue.